app/frontend/tokenRegistry/tokenRegistryApi.ts

```diff
@@ -0,0 +1,46 @@
+import request from '../wallet/helpers/request'
+import {ADALITE_CONFIG} from '../config'
+import {RegisteredTokenMetadata} from '../types'
+import {TokenRegistryApi} from './types'
+
+export default (): TokenRegistryApi => {
+  const fetchTokensMetadata = (subjects: string[]): Promise<any> => {
+    const url = `${ADALITE_CONFIG.ADALITE_SERVER_URL}/api/tokenRegistry/getTokensMetadata`
+    const requestBody = {subjects}
+    return request(url, 'POST', JSON.stringify(requestBody), {
+      'Content-Type': 'application/json',
+    })
+  }
+
+  const parseTokensMetadata = (toParse: any): {[subject: string]: RegisteredTokenMetadata} => {
+    if (toParse?.Right) {
+      return toParse.Right.reduce((acc, tokenMetadata) => {
+        const {subject, description, ticker, url, logoHex} = tokenMetadata
+        acc[subject] = {subject, description, ticker, url, logoHex}
+        acc[tokenMetadata.subject] = {
+          subject: tokenMetadata.subject,
+          description: tokenMetadata.description.value,
+          ticker: tokenMetadata?.ticker?.value,
+          url: tokenMetadata?.url?.value,
+          logoHex: tokenMetadata?.logo?.value,
+          decimals: tokenMetadata?.decimals?.value,
+        }
+        return acc
+      }, {})
+    } else {
+      return {}
+    }
+  }
+
+  const getTokensMetadata = async (
```

Comment:
If I understood correctly, this function we currently don't cache the token metadata in the fronted when we reload the wallet even though the tokens of the user didn't change. I'd consider wrapping the fetchTokensMetadata in some memorization to prevent such redundant re-fetches of data: https://lodash.com/docs/4.17.15#memoize

Reply:
Fixed